src/random_test.cpp

```diff
@@ -2608,11 +2608,16 @@ void alter_tablespace_rename(Thd1 *thd) {
 }
 
 /* load special sql from a file */
-static std::vector<std::string> load_special_sql_from() {
+static std::vector<std::string> load_grammar_sql_from() {
   std::vector<std::string> array;
-  auto file = opt_string(SQL_FILE);
-  std::string sql;
+  auto grammar_file = opt_string(GRAMMAR_FILE);
+  std::string sql,file;
 
+  if (grammar_file == "grammar.sql")
+    file = fullpath + grammar_file;
+  else
+    file = grammar_file;
```

Comment:
It seems this logic may not work when the file name in `grammar.sql` and if it is placed in a different directory other than the pstress binary directory.

Reply:
The logic is correct. When the user does not pass --grammar-file option, only then the value of grammar_file defaults to "grammar.sql"
In all other cases, the value of grammar_file will be equal to:
"/some/random/path/grammar.sql"
"some/random/path/custom.sql"
In both scenatio the if condition will return false

Reply:
Ok. Got it.